src/main/java/org/radarcns/android/device/AbstractDeviceManager.java

```diff
@@ -86,17 +89,38 @@ protected void updateStatus(DeviceStatusListener.Status status) {
      * override with an empty implementation.
      */
     protected void registerDeviceAtReady() {
-        service.registerDevice(deviceName, null);
+        service.registerDevice(deviceName, Collections.<String, String>emptyMap());
+    }
+
+    /**
+     * Creates and registers an Avro topic
+     * @param name The name of the topic
+     * @param valueClass The value class
+     * @param <V>
+     * @return
+     */
+    protected <V extends SpecificRecord> AvroTopic<ObservationKey, V> createTopic(String name, Class<V> valueClass) {
+        try {
+            Method method = valueClass.getMethod("getClassSchema");
+            Schema valueSchema = (Schema) method.invoke(null);
+            AvroTopic<ObservationKey, V> topic = new AvroTopic<>(
+                    name, ObservationKey.getClassSchema(), valueSchema, ObservationKey.class, valueClass);
+            dataHandler.registerTopic(topic);
+            return topic;
+        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | IOException e) {
+            logger.error("Error creating topic " + name, e);
+            throw new RuntimeException(e);
+        }
```

Comment:
Android strongly suggests against using reflection. This was the reason for the verbose old-style createTopic (DeviceTopics). See for example
http://blog.nimbledroid.com/2016/02/23/slow-Android-reflection.html
where a single reflection call took about 81 ms. If we have about 25 topics, that would cause an additional startup delay of about 2 seconds.